### ISSUE TITLE
Compiling zig.h with GCC in a Zephyr project throws a ton of errors and warnings

### ISSUE THREAD

**Issue by BeckmaR** [ziglang/zig#16676]
### Zig Version

0.10.1

### Steps to Reproduce and Observed Behavior

I am currently trying to integrate Zig with Zephyr by building a very minimal static library with Zig and linking my Zephyr project against it. I am building for `x86_64-freestanding` or `qemu_x86_64` in Zephyr. This works perfectly fine if I declare my exported function myself (for the moment, as it's fairly simple), but I hit problems when I try to use `-femit-h` and use the emitted header, because then the Zephyr toolchain tries to compile `zig.h`. This produces numerous errors and I don't really understand what causes this and how I could even start to fix it.

First, here is the complete call from `compile_commands.json`:
```
{
  "directory": "/Users/rene/git/zig-zephyr/zig-zephyr-test/build",
  "command": "/Users/rene/zephyr-sdk-0.15.0/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc -DKERNEL -D__ZEPHYR__=1 -I/Users/rene/git/zig-zephyr/zig-zephyr-test/app/src -I/opt/homebrew/Cellar/zig/0.10.1/lib/zig -I/Users/rene/git/zig-zephyr/zephyr/include -I/Users/rene/git/zig-zephyr/zig-zephyr-test/build/zephyr/include/generated -I/Users/rene/git/zig-zephyr/zephyr/soc/x86/ia32 -I/Users/rene/git/zig-zephyr/zig-zephyr-test/include -isystem /Users/rene/git/zig-zephyr/zephyr/lib/libc/minimal/include -isystem /Users/rene/zephyr-sdk-0.15.0/x86_64-zephyr-elf/bin/../lib/gcc/x86_64-zephyr-elf/12.1.0/include -isystem /Users/rene/zephyr-sdk-0.15.0/x86_64-zephyr-elf/bin/../lib/gcc/x86_64-zephyr-elf/12.1.0/include-fixed -m32  -fno-strict-aliasing -Os -imacros /Users/rene/git/zig-zephyr/zig-zephyr-test/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -m32 -msoft-float -Wa,--divide --sysroot=/Users/rene/zephyr-sdk-0.15.0/x86_64-zephyr-elf/x86_64-zephyr-elf -imacros /Users/rene/git/zig-zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/Users/rene/git/zig-zephyr/zig-zephyr-test/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/Users/rene/git/zig-zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/Users/rene/git/zig-zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -mpreferred-stack-boundary=2 -mno-mmx -mno-sse -std=c99 -nostdinc -Wno-error -o CMakeFiles/app.dir/src/main.c.obj -c /Users/rene/git/zig-zephyr/zig-zephyr-test/app/src/main.c",
  "file": "/Users/rene/git/zig-zephyr/zig-zephyr-test/app/src/main.c"
},
```

Here is an excerpt of the errors I am seeing:
```
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h:1256:99: error: request for member 'hi' in something not a structure or union
 1256 |     return zig_sub_i128(zig_div_trunc_i128(lhs, rhs), (((lhs.hi ^ rhs.hi) & zig_rem_i128(lhs, rhs).hi) < zig_as_i64(0)) ? zig_as_i128(0, 1) : zig_as_i128(0, 0));
      |                                                                                                   ^
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h:1256:25: error: incompatible type for argument 1 of 'zig_sub_i128'
 1256 |     return zig_sub_i128(zig_div_trunc_i128(lhs, rhs), (((lhs.hi ^ rhs.hi) & zig_rem_i128(lhs, rhs).hi) < zig_as_i64(0)) ? zig_as_i128(0, 1) : zig_as_i128(0, 0));
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         |
      |                         int
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h:1249:46: note: expected 'zig_i128' but argument is of type 'int'
 1249 | static inline zig_i128 zig_sub_i128(zig_i128 lhs, zig_i128 rhs) {
      |                                     ~~~~~~~~~^~~
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h: In function 'zig_mod_i128':
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h:1260:20: error: invalid initializer
 1260 |     zig_i128 rem = zig_rem_i128(lhs, rhs);
      |                    ^~~~~~~~~~~~
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h:1261:16: error: invalid operands to binary + (have 'zig_i128' and 'zig_i128')
 1261 |     return rem + (((lhs.hi ^ rhs.hi) & rem.hi) < zig_as_i64(0) ? rhs : zig_as_i128(0, 0));
      |                ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                      |
      |                                                                      zig_i128
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h: In function 'zig_shr_i128':
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h:1286:81: error: wrong type argument to unary minus
 1286 |     zig_i128 sign_mask = zig_cmp_i128(lhs, zig_as_i128(0, 0)) < zig_as_i32(0) ? -zig_as_i128(0, 1) : zig_as_i128(0, 0);
      |                                                                                 ^
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h: In function 'zig_shlw_i128':
/opt/homebrew/Cellar/zig/0.10.1/lib/zig/zig.h:1100:57: error: request for member 'hi' in something not a structure or union
 1100 | #define zig_bitcast_u128(val) zig_as_u128((zig_u64)(val).hi, (val).lo)
```

As stated, I have no idea what causes this funny cascade of errors. 

For reference, here are my files:
main.c
```
#include <zephyr/kernel.h>
#include <app_version.h>
#include "lib.h"

#include <zephyr/logging/log.h>
LOG_MODULE_REGISTER(main, CONFIG_APP_LOG_LEVEL);

int main(void)
{
	int val;

	printk("Zephyr Example Application %s\n", APP_VERSION_STRING);

	val = add(40, 2);

	printk("Calculated value: %d\n", val);

	return 0;
}
```

lib.zig
```
const std = @import("std");

export fn add(a: i32, b: i32) i32 {
    return a + b;
}
```

the generated lib.h
```
#include "zig.h"
zig_extern zig_i32 add(zig_i32 const a0, zig_i32 const a1);
```

### Expected Behavior

I would like the `zig.h` header to be consumed without errors ;) I'm also wondering if this header is really needed? I'm thinking that all the zig internals are included in the static library `lib.a`, so the only thing I'd need is a header with the zig types, like `zig_types.h` defining `zig_i32`, not all the functions causing the errors.

I have to state I'm very new to Zig and I'm probably doing something fundamentally wrong.

**Comment by jacobly0** [comment]
Fixed forever ago, example: #14002

**Comment by BeckmaR** [comment]
@jacobly0 Sorry, my fault, I should have checked if there is a newer version of zig available. I only checked homebrew, where 0.10.1 is the newest version. I will try with 0.11.0.
